backend/app/utils.py

```
@@ -1238,45 +1374,223 @@ def get_metadata(portal_model_instance, credential_token=None):
+        if "dataSource" in data_structure:
+            data_sources = data_structure.get("dataSource", {}).get("dataSources", {})
+            if isinstance(data_sources, dict):
+                for ds_key, ds_value in data_sources.items():
+                    if isinstance(ds_value, dict):
+                        try:
+                            ds_item_id = ds_value.get("itemId")
+                            if ds_item_id and ds_item_id not in itemids:
+                                path = f"dataSource.dataSources.{ds_key}"
+                                extracted_resources.append([ds_item_id, path, "itemId", "datasource", "datasource"])
+                                itemids.append(ds_item_id)
+                        except:
```

Ruff (0.14.10)

Do not use bare `except`

(E722)